pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -95,108 +64,69 @@ func (c *cyclonedxParser) GetIdentities(ctx context.Context) []common.TrustInfor
 	return nil
 }
 
-// TODO(bulldozer): replace with GetPredicates
-// func (c *cyclonedxParser) CreateEdges(ctx context.Context, foundIdentities []common.TrustInformation) []assembler.GuacEdge {
-// 	edges := []assembler.GuacEdge{}
-// 	visited := make(map[string]bool)
-// 	addEdges(c.rootComponent, &edges, visited)
-// 	return edges
-// }
-
-func (c *cyclonedxParser) addRootPackage(cdxBom *cdx.BOM) {
-	// oci purl: pkg:oci/debian@sha256%3A244fd47e07d10?repository_url=ghcr.io/debian&tag=bullseye
+func (c *cyclonedxParser) getTopLevelPackage(cdxBom *cdx.BOM) error {
 	if cdxBom.Metadata.Component != nil {
-		rootPackage := assembler.PackageNode{}
-		rootPackage.Name = cdxBom.Metadata.Component.Name
-		rootPackage.NodeData = *assembler.NewObjectMetadata(c.doc.SourceInformation)
-		if cdxBom.Metadata.Component.PackageURL != "" {
-			rootPackage.Purl = cdxBom.Metadata.Component.PackageURL
-			rootPackage.Version = cdxBom.Metadata.Component.Version
-			rootPackage.Tags = []string{string(cdxBom.Metadata.Component.Type)}
-		} else {
+		purl := cdxBom.Metadata.Component.PackageURL
+		if cdxBom.Metadata.Component.PackageURL == "" {
 			if cdxBom.Metadata.Component.Type == cdx.ComponentTypeContainer {
 				splitImage := strings.Split(cdxBom.Metadata.Component.Name, "/")
-				if len(splitImage) == 3 {
-					// example: gcr.io/distroless/static:nonroot
-					splitTag := strings.Split(splitImage[2], ":")
-					if len(splitTag) == 2 {
-						rootPackage.Purl = "pkg:oci/" + splitTag[0] + "@" + cdxBom.Metadata.Component.Version +
-							"?repository_url=" + splitImage[0] + "/" + splitImage[1] + "/" + splitTag[0] + "&tag=" + splitTag[1]
-					} else {
-						// no tag specified
-						rootPackage.Purl = "pkg:oci/" + splitImage[2] + "@" + cdxBom.Metadata.Component.Version +
-							"?repository_url=" + splitImage[0] + "/" + splitImage[1] + "/" + splitImage[2] + "&tag="
-					}
-				} else if len(splitImage) == 2 {
-					// example: library/debian:latest
-					splitTag := strings.Split(splitImage[1], ":")
-					if len(splitTag) == 2 {
-						rootPackage.Purl = "pkg:oci/" + splitTag[0] + "@" + cdxBom.Metadata.Component.Version +
-							"?repository_url=" + splitImage[0] + "/" + splitTag[0] + "&tag=" + splitTag[1]
-					} else {
-						// no tag specified
-						rootPackage.Purl = "pkg:oci/" + splitImage[1] + "@" + cdxBom.Metadata.Component.Version +
-							"?repository_url=" + splitImage[0] + "/" + splitImage[1] + "&tag="
-					}
+
+				const (
+					ociPrefix        = "pkg:oci/"
+					repositoryURLKey = "?repository_url="
+					tagKey           = "&tag="
+				)
+
+				splitTag := strings.Split(splitImage[len(splitImage)-1], ":")
```

Comment:
also we can pass in just the document name `cdxBom.Metadata.Component.Name` (similar to SPDX)